File: problems/2211_count_collisions_on_a_road/2211_count_collisions_on_a_road.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        n = len(directions)

        i = 0
        while i < n and directions[i] == 'L':
            i += 1
        
        j = n - 1
        while j >= 0 and directions[j] == 'R':
            j -= 1
        
        collisions = 0
        for k in range(i, j + 1):
            if directions[k] != 'S':
                collisions += 1

        return collisions

File: problems/2211_count_collisions_on_a_road/2211_count_collisions_on_a_road.java
class Solution {
    public int countCollisions(String directions) {
        int n = directions.length();
        char[] chars = directions.toCharArray();

        int left = 0;
        while (left < n && chars[left] == 'L')
            left++;

        int right = n - 1;
        while (right >= 0 && chars[right] == 'R')
            right--;

        int collisions = 0;
        for (int i = left; i <= right; i++) {
            if (chars[i] != 'S')
                collisions++;
        }
        return collisions;
    }
}